cfme/tests/automate/test_service_automate.py

```diff
@@ -209,3 +209,139 @@ def test_automate_method_with_dialog(request, appliance, catalog, setup_dynamic_
         provision_request = service_catalogs.order()
         provision_request.wait_for_request()
         request.addfinalizer(provision_request.remove_request)
+
+
+@pytest.fixture(scope="function")
+def copy_klass(domain):
+    """This fixture copies ServiceProvision_Template which is required while selecting instance in
+    catalog item"""
+    domain.parent.instantiate(name="ManageIQ").namespaces.instantiate(
+        name="Service"
+    ).namespaces.instantiate(name="Provisioning").namespaces.instantiate(
+        name="StateMachines"
+    ).classes.instantiate(
+        name="ServiceProvision_Template"
+    ).copy_to(
+        domain.name
+    )
+
+    klass = (
+        domain.namespaces.instantiate(name="Service")
+        .namespaces.instantiate(name="Provisioning")
+        .namespaces.instantiate(name="StateMachines")
+        .classes.instantiate(name="ServiceProvision_Template")
+    )
+    yield klass
```

Comment:
Hey @mshriver 
- Actually, this ```klass``` comes under custom domain which is global fixture. So whenever we delete domain; this klass gets deleted. Hence I did not add cleanup here.
- But I agree with you here because I noticed that if I copied this same class for more test cases then it creates trouble and fails to copy same class again with saying klass already exists. So from now, I am adding cleanup for all the child items under domain